sn_transfers/src/cashnotes/transaction.rs

```diff
@@ -132,44 +132,50 @@ impl Transaction {
             })?;
 
         if input_sum != output_sum {
-            Err(Error::InconsistentTransaction)
+            Err(Error::UnbalancedTransaction)
         } else {
             Ok(())
         }
     }
 
-    /// Verifies a transaction including signed spends.
+    /// Verifies a transaction with Network held signed spends.
     ///
-    /// This function relies/assumes that the caller (wallet/client) obtains
-    /// the Transaction (held by every input spend's close group) in a
-    /// trustless/verified way. I.e., the caller should not simply obtain a
-    /// spend from a single peer, but must get the same spend from all in the close group.
+    /// This function assumes that the signed spends where previously fetched from the Network and where not double spent.
+    /// This function will verify that:
+    /// - the transaction is balanced (sum inputs = sum outputs)
+    /// - the inputs and outputs are unique
+    /// - the inputs and outputs are different
+    /// - the inputs have a corresponding signed spend
+    /// - those signed spends are valid and refer to this transaction
     pub fn verify_against_inputs_spent(&self, signed_spends: &BTreeSet<SignedSpend>) -> Result<()> {
         // verify that the tx has at least one input
         if self.inputs.is_empty() {
             return Err(Error::MissingTxInputs);
         }
 
-        // check if we have spends for all inputs
-        if signed_spends.is_empty() {
-            return Err(Error::MissingTxInputs)?;
-        }
-        if signed_spends.len() != self.inputs.len() {
-            return Err(Error::SignedSpendInputLenMismatch {
-                got: signed_spends.len(),
-                expected: self.inputs.len(),
-            });
```

Comment:
This check 
` if signed_spends.len() != self.inputs.len()`
along with the one L184
`.any(|m| m.unique_pubkey == *signed_spend.unique_pubkey())` 
is flawed as it doesn't prevent one from using multiple identical valid `signed_spends`. 
Eg: say I have input A and B. The spend for B is invalid (someone else's money for example), so I provide twice input A (which is valid). Both checks pass. The Tx is marked as valid.